vendor/k8s.io/gengo/parser/parse.go

```diff
@@ -749,3 +749,14 @@ func (b *Builder) addVariable(u types.Universe, useName *types.Name, in *tc.Var)
 	out.Underlying = b.walkType(u, nil, in.Type())
 	return out
 }
+
+// importPathToFullPackage takes an import path and returns the actual package
+// it doesn't work for any folder actually called `vendor` (we don't dep any today),
+// so hopefully we keep up the streak
+func importPathToFullPackage(importPath string) string {
+	if !strings.Contains(importPath, "/vendor/") {
```

Comment:
This works because we only vendor at the root.  I bet this will break in contrib, which vendors in subdirs.  Curiously we make the same assumption in using CWD as the relative root, so I can't fault this for being broken.

What I was thinking is more like saving canon path but when we emit imports, strip the common pfx.

e.g. when emitting for pkg `k8s.io/kubernetes/pkg/api` and you find an import for `k8s.io/kubernetes/vendor/github.com/foo/bar`, we recognize `k8s.io/kubernetes/vendor/` as the vendor-root, and `k8s.io/kubernetes/` happens to be a prefix of the package we are emitting into, so we can strip the vendor-root => github.com/foo/bar

e.g. for `k8s.io/contrib/some/package` if we find an import for `k8s.io/contrib/some/vendor/github.com/foo/bar` we detect `k8s.io/contrib/some/vendor/` as vendor-root and `k8s.io/contrib/some` is a prefix of the target pkg, so we can strip the vendor-root => github.com/foo/bar

e.g. for `k8s.io/contrib//other/package` if we find an import for `k8s.io/contrib/vendor/github.com/foo/bar` we detect `k8s.io/contrib/vendor/` as vendor-root and `k8s.io/contrib` is a prefix of the target pkg, so we can strip the vendor-root => github.com/foo/bar